bin/liquidator/src/trigger_tcs.rs

```diff
@@ -57,15 +72,16 @@ fn tcs_has_plausible_premium(
     Ok(cost <= premium)
 }
 
-async fn tcs_is_interesting(
-    mango_client: &MangoClient,
+fn tcs_is_interesting(
+    context: &MangoGroupContext,
+    account_fetcher: &chain_data::AccountFetcher,
     tcs: &TokenConditionalSwap,
     token_swap_info: &token_swap_info::TokenSwapInfoUpdater,
     now_ts: u64,
 ) -> anyhow::Result<bool> {
-    Ok(!tcs.is_expired(now_ts)
-        && tcs_is_in_price_range(mango_client, tcs).await?
-        && tcs_has_plausible_premium(tcs, token_swap_info)?)
+    Ok(tcs.is_expired(now_ts)
```

Comment:
ah, now you consider expired ones also, because they can be evicted and tcs slot can be made free from the mango account, or?

Reply:
Yeah, the definition of "interesting" changed here. :P

I collect all tcs that are possible good to execute or expired, put their volume in the list and pick one. Then the execution function either closes or triggers.